src/app/backend/replicasetdetail.go

```diff
@@ -211,3 +229,41 @@ func getRestartCount(pod api.Pod) int {
 	}
 	return restartCount
 }
+
+// Returns internal endpoint name for the given service properties, e.g.,
+// "my-service.namespace 80/TCP" or "my-service 53/TCP,53/UDP".
+func getInternalEndpoint(serviceName string, namespace string, ports []api.ServicePort) Endpoint {
+
+	name := serviceName
+	if namespace != api.NamespaceDefault {
+		name = name + "." + namespace
```

Comment:
Very good practice is not using ```+``` for string concatenation.
Strings are immutable in Go - each time a string is assigned to a variable a new address is assigned in memory to represent the new value.  As the strings you are manipulating become large this process becomes increasingly slow. In fact, this method of naive appending is O(N2). (The same is in Java so StringBuffer is used instead ```+=```)
Here you should also use Buffer:
```
	bufferName := bytes.NewBufferString(name)
	bufferName.WriteString(".")
    bufferName.WriteString(namespace)
	name = bufferName.String()
```